### DISCUSSION TITLE
poetry environment python version

### DISCUSSION THREAD

**Discussion by 0xdolan** [python-poetry/poetry#7863] [Q&A]
The Python version specified in the `pyproject.toml` file is not being used when generating a poetry project on Docker (Ubuntu). Despite specifying the version as `^3.11.0`, it is using `Python 3.11.0rc1`, which is a different version. Additionally, there are some vulnerabilities shown by the Docker vulnerability checks.

What is the best practice to ensure that the installed Python version in the environment matches the version specified in the pyproject.toml file? Also, please take care of the security CVE issues.

Please refer to the following screenshots for more information:

`pyproject.toml` python version:
![Screenshot 2023-05-02 154702](https://user-images.githubusercontent.com/18519747/235675364-a94a2601-82fb-4dce-b54a-6548f5f69c09.png)

`python version in poetry environment`:
![Screenshot 2023-05-02 154339](https://user-images.githubusercontent.com/18519747/235679109-457e4043-6899-451e-9edf-ca7e962e682d.png)

`docker CVE issues`:
![Screenshot 2023-05-02 151052](https://user-images.githubusercontent.com/18519747/235675426-90f2fb17-3f75-4433-b569-8df12535ac04.png)

Thank you.

**Comment by dimbleby** [comment]
If you don't want to use python 3.11.0rc1 - then use a docker image that doesn't use python 3.11.0rc1.  

> please take care of the security CVE issues.

These are not poetry's issues: they are CVEs against the old version of python that you are using.  _You_ should take care of these CVEs - again, by using a docker image containing a more recent python.

**Comment by 0xdolan** [comment]
Thank you. There are lots of different ways to resolve the issue with using the python version. I just wanted to know if the version the poetry virtual environment is generated is stand alone or it is dependent to the installed python globally. If it is dependent to the system wide python version, why it should use the rc version. That's why I asked it here to have some information on the process.

**Reply by radoering** [reply]
You can tell poetry, which Python version should be used (as long as it is allowed by the project's python constraint): https://python-poetry.org/docs/managing-environments/#switching-between-environments

Poetry does not bring its own Python. It can only use what's available on the system.
